Core/src/Utilities/SpacePointUtility.cpp

```diff
@@ -0,0 +1,339 @@
+  // Calculate the scaling factor to project lengths of the second SDE on the
+  // first SDE
+  double secOnFirstScale =
+      spParams.firstBtmToTop.dot(spParams.secondBtmToTop) /
+      (spParams.mag_firstBtmToTop * spParams.mag_firstBtmToTop);
+  // Check if both overshoots are in the same direction
+  if (spParams.m > 1. && spParams.n > 1.) {
+    // Calculate the overshoots
+    double mOvershoot = spParams.m - 1.;
+    double nOvershoot =
+        (spParams.n - 1.) * secOnFirstScale;  // Perform projection
+    // Resolve worse overshoot
+    double biggerOvershoot = std::max(mOvershoot, nOvershoot);
+    // Move m and n towards 0
+    spParams.m -= biggerOvershoot;
+    spParams.n -= (biggerOvershoot / secOnFirstScale);
+    // Check if this recovered the space point
+
+    if (fabs(spParams.m) < spParams.limit && fabs(spParams.n) < spParams.limit)
+      return Result<void>::success();
+    else
+      return Result<void>::failure(m_error);
+  }
+  // Check if both overshoots are in the same direction
+  if (spParams.m < -1. && spParams.n < -1.) {
+    // Calculate the overshoots
+    double mOvershoot = -(spParams.m + 1.);
+    double nOvershoot =
+        -(spParams.n + 1.) * secOnFirstScale;  // Perform projection
+    // Resolve worse overshoot
+    double biggerOvershoot = std::max(mOvershoot, nOvershoot);
+    // Move m and n towards 0
+    spParams.m += biggerOvershoot;
+    spParams.n += (biggerOvershoot / secOnFirstScale);
+    // Check if this recovered the space point
+    if (fabs(spParams.m) < spParams.limit && fabs(spParams.n) < spParams.limit)
+      return Result<void>::success();
+  }
+  // No solution could be found
+  return Result<void>::failure(m_error);
+}
+
+Result<double> SpacePointUtility::calcPerpendicularProjection(
+    const std::pair<Vector3, Vector3>& stripEnds1,
+    const std::pair<Vector3, Vector3>& stripEnds2,
+    SpacePointParameters& spParams) const {
+  /// This approach assumes that no vertex is available. This option aims to
+  /// approximate the space points from cosmic data.
+  /// The underlying assumption is that the best point is given by the  closest
+  /// distance between both lines describing the SDEs.
+  /// The point x on the first SDE is parametrized as a + lambda0 * q with  the
+  /// top end a of the strip and the vector q = a - b(ottom end of the  strip).
+  /// An analogous parametrization is performed of the second SDE with y = c  +
+  /// lambda1 * r.
+  /// x get resolved by resolving lambda0 from the condition that |x-y| is  the
+  /// shortest distance between two skew lines.
+
+  spParams.firstBtmToTop = stripEnds1.first - stripEnds1.second;
+  spParams.secondBtmToTop = stripEnds2.first - stripEnds2.second;
+
+  Vector3 ac = stripEnds2.first - stripEnds1.first;
+  double qr = (spParams.firstBtmToTop).dot(spParams.secondBtmToTop);
+  double denom = spParams.firstBtmToTop.dot(spParams.firstBtmToTop) - qr * qr;
+  // Check for numerical stability
+  if (fabs(denom) > 10e-7) {
```

Comment:
```suggestion
  if (fabs(denom) > 1e-6) {
```

This also occurs in `Core/include/Acts/Digitization/detail/DoubleHitSpacePointBuilder.ipp`